e2e/next-sandbox/test/comments/index.test.ts

```diff
@@ -22,21 +22,25 @@ test.describe("Comments", () => {
   test.afterEach(() => Promise.all(pages.map((page) => page.close())));
 
   test("verify A and B display same number of threads after threads are loaded", async () => {
-    await waitForJson(pages, "#isLoading", false);
-    await waitUntilEqualOnAllPages(pages, "#numOfThreads");
+    await waitForJson(pages, "#isLoading", false, { timeout: 15_000 });
+    await waitUntilEqualOnAllPages(pages, "#numOfThreads", { interval: 250 });
   });
 
   test("verify thread creation on B is broadcasted correctly to A", async () => {
     const [page1, page2] = pages;
 
-    await waitForJson(pages, "#isLoading", false);
+    await waitForJson(pages, "#isLoading", false, { timeout: 15_000 });
+    await waitUntilEqualOnAllPages(pages, "#numOfThreads", { interval: 250 });
```

Comment:
This waitUntilEqualOnAllPages sync step is crucial, before reading the initial `n` value on the line below. If you read it before you've ensured they are in sync, then the initial value could be different, which causes the rest of the computation to get weird.